src/analysis/normed_space/normed_group_hom.lean

```diff
@@ -0,0 +1,461 @@
+structure normed_group_hom (V W : Type*) [normed_group V] [normed_group W] :=
+(to_fun : V → W)
+(map_add' : ∀ v₁ v₂, to_fun (v₁ + v₂) = to_fun v₁ + to_fun v₂)
+(bound' : ∃ C, ∀ v, ∥to_fun v∥ ≤ C * ∥v∥)
+
+/-- Associate to a group homomorphism a bounded group homomorphism under a norm control condition.
+-/
+def add_monoid_hom.mk_continuous {V W : Type*} [normed_group V] [normed_group W] (f : V →+ W)
+  (C : ℝ) (h : ∀ v, ∥f v∥ ≤ C * ∥v∥) : normed_group_hom V W :=
+{ bound' := ⟨C, h⟩, ..f }
+
+lemma exists_pos_bound_of_bound {V W : Type*} [normed_group V] [normed_group W]
+  {f : V → W} (M : ℝ) (h : ∀x, ∥f x∥ ≤ M * ∥x∥) :
+  ∃ N, 0 < N ∧ ∀x, ∥f x∥ ≤ N * ∥x∥ :=
+⟨max M 1, lt_of_lt_of_le zero_lt_one (le_max_right _ _), λx, calc
+  ∥f x∥ ≤ M * ∥x∥ : h x
+  ... ≤ max M 1 * ∥x∥ : mul_le_mul_of_nonneg_right (le_max_left _ _) (norm_nonneg _) ⟩
+
+namespace normed_group_hom
+
+variables {V V₁ V₂ V₃ : Type*}
+variables [normed_group V] [normed_group V₁] [normed_group V₂] [normed_group V₃]
+variables {f g : normed_group_hom V₁ V₂}
+
+instance : has_coe_to_fun (normed_group_hom V₁ V₂) := ⟨_, normed_group_hom.to_fun⟩
+
+initialize_simps_projections normed_group_hom (to_fun → apply)
+
+lemma coe_inj (H : ⇑f = g) : f = g :=
+by cases f; cases g; congr'; exact funext H
+
+lemma coe_injective : @function.injective (normed_group_hom V₁ V₂) (V₁ → V₂) coe_fn :=
+by apply coe_inj
+
+lemma coe_inj_iff : f = g ↔ ⇑f = g := ⟨congr_arg _, coe_inj⟩
+
+@[ext] lemma ext (H : ∀ x, f x = g x) : f = g := coe_inj $ funext H
+
+lemma ext_iff : f = g ↔ ∀ x, f x = g x := ⟨by rintro rfl x; refl, ext⟩
+
+variables (f g)
+
+@[simp] lemma to_fun_eq_coe : f.to_fun = f := rfl
+
+@[simp] lemma coe_mk (f) (h₁) (h₂) (h₃) : ⇑(⟨f, h₁, h₂, h₃⟩ : normed_group_hom V₁ V₂) = f := rfl
+
+/-- The group homomorphism underlying a bounded group homomorphism. -/
+def to_add_monoid_hom (f : normed_group_hom V₁ V₂) : V₁ →+ V₂ :=
+add_monoid_hom.mk' f f.map_add'
+
+@[simp] lemma coe_to_add_monoid_hom : ⇑f.to_add_monoid_hom = f := rfl
+
+lemma to_add_monoid_hom_injective :
+  function.injective (@normed_group_hom.to_add_monoid_hom V₁ V₂ _ _) :=
+λ f g h, coe_inj $ show ⇑f.to_add_monoid_hom = g, by { rw h, refl }
+
+@[simp] lemma mk_to_add_monoid_hom (f) (h₁) (h₂) :
+  (⟨f, h₁, h₂⟩ : normed_group_hom V₁ V₂).to_add_monoid_hom = add_monoid_hom.mk' f h₁ := rfl
+
+@[simp] lemma map_zero : f 0 = 0 := f.to_add_monoid_hom.map_zero
+
+@[simp] lemma map_add (x y) : f (x + y) = f x + f y := f.to_add_monoid_hom.map_add _ _
+
+@[simp] lemma map_sum {ι : Type*} (v : ι → V₁) (s : finset ι) :
+  f (∑ i in s, v i) = ∑ i in s, f (v i) :=
+f.to_add_monoid_hom.map_sum _ _
+
+@[simp] lemma map_sub (x y) : f (x - y) = f x - f y := f.to_add_monoid_hom.map_sub _ _
+
+@[simp] lemma map_neg (x) : f (-x) = -(f x) := f.to_add_monoid_hom.map_neg _
+
+/-- Make a normed group hom from a group hom and a norm bound. -/
+def mk' (f : V₁ →+ V₂) (C : ℝ≥0) (hC : ∀ x, ∥f x∥ ≤ C * ∥x∥) : normed_group_hom V₁ V₂ :=
```

Comment:
This seems to be almost the same thing as `add_monoid_hom.mk_continuous`, which is better from the point of view of namespacing (with the difference that the bound is now in nnreal). Can you rename this to `add_monoid_hom.mk_continuous'`?

Reply:
What do you think of `add_monoid_hom.mk_bounded`?
Which also brings back a question that I don't really have a strong answer for: should I rename of `normed_group_hom` to `bounded_group_hom` everywhere?

Reply:
I am fine with `normed_group_hom`, so do as you prefer. You can also have `add_monoid_hom.mk_bounded` and `add_monoid_hom.mk_bounded'`, or `add_monoid_hom.mk_continuous` and `add_monoid_hom.mk_continuous'`, as you prefer as long as the names match.

Reply:
Sorry, I couldn't find `add_monoid_hom.mk_continuous`.
I've now renamed `mk'` to `add_monoid_hom.mk_normed_group_hom`, which is as unambiguous as it gets, I think.

Reply:
It's on line 35.

Reply:
:facepalm: